src/webapi/indexeddb.rs

```diff
@@ -0,0 +1,1145 @@
+#[reference(instance_of = "IDBObjectStore")]
+pub struct IDBObjectStore( Reference );
+
+impl IDBObjectStoreIndexSharedMethods for IDBObjectStore {}
+
+impl IDBObjectStore {
+       
+    /// The index_names read-only property of the `IDBObjectStore` interface returns a list of th
+    /// names of indexes on objects in this object store.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/indexNames)
+    pub fn index_names( &self ) -> DOMStringList {
+        js! ( return @{self}.indexNames ).try_into().unwrap()
+    }
+    
+    /// The `IDBTransaction` object to which this object store belongs.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/transaction)
+    pub fn transaction( &self ) -> IDBTransaction {
+        js! (
+            return @{self.as_ref()}.transaction;
+        ).try_into().unwrap()
+    }
+    
+    /// Returns the value of the auto increment flag for this object store.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/autoIncrement)
+    pub fn auto_increment( &self ) -> bool {
+        js! (
+            return @{self.as_ref()}.autoIncrement;
+        ).try_into().unwrap()
+    }
+    
+    /// Updates a given record in a database, or inserts a new record if the given item does not already exist.
+    /// The key is only needed if 
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/put)
+    pub fn put<T: Into<Option<Value>>>( &self, value: Value, key: T) -> Result<IDBRequest, IDBUpdateError> {
+        match key.into() {
+            None => js_try! (
+                return @{self.as_ref()}.put(@{value.as_ref()});
+            ),
+            Some(key) => js_try! (
+                return @{self.as_ref()}.put(@{value.as_ref()}, @{key.as_ref()});
+            )
+        }.unwrap()
+    }
+    
+    /// Returns an `IDBRequest` object, and, in a separate thread, creates a structured clone of the value, and stores the cloned value in the object store. This is for adding new records to an object store.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/add)
+    pub fn add<T: Into<Option<Value>>>( &self, value: Value, key: T) -> Result<IDBRequest, IDBAddError> {
+        match key.into() {
+            None => js_try! (
+                return @{self.as_ref()}.add(@{value.as_ref()});
+            ),
+            Some(key) => js_try! (
+                return @{self.as_ref()}.add(@{value.as_ref()}, @{key.as_ref()});
+            )
+        }.unwrap()
+    }
+
+    /// returns an `IDBRequest` object, and, in a separate thread, deletes the specified record or records.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/delete)
+    pub fn delete( &self, query: Value) -> Result<IDBRequest, IDBObjectStoreDeleteError> {
```

Comment:
The other methods (e.g. `get_all`) all take `Q: Into<IDBKeyOrKeyRange>` as `query`, and yet here we take `Value`. Shouldn't it be consistent?

Reply:
How would `Q: Into<IDBKeyOrKeyRange>` convert to `Value` here? Attempted to use `.into()` but this fails:

```diff
--- a/src/webapi/indexeddb.rs
+++ b/src/webapi/indexeddb.rs
@@ -884,7 +884,8 @@ impl IDBObjectStore {
     /// Returns an `IDBRequest` object, and, in a separate thread, deletes the specified record or records.
     ///
     /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/delete)
-    pub fn delete( &self, query: Value) -> Result<IDBRequest, IDBObjectStoreDeleteError> {
+    pub fn delete<Q: Into<IDBKeyOrKeyRange>>( &self, query: Q) -> Result<IDBRequest, IDBObjectStoreDeleteError> {
+        let query: Value = query.into();
         js_try! (
             return @{self.as_ref()}.delete(@{query.as_ref()});
         ).unwrap()
```